### PR TITLE
Add support for declare export + fixes

### PR DESCRIPTION
**Pull request by danez**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | yes
| Deprecations?     | no
| Spec compliancy?  | yes
| Tests added/pass? | yes
| Fixed tickets     | #74
| License           | MIT

This add support for `declare export`. I tried to follow the flow parser as close as possible.
I also found some related problems and fixed them.
* ObjectTypeProperties were missing the field `kind`
* getters and setters were not allowed in TypeClasses/interfaces

### CODE REVIEW

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node v4 - linux](https://travis-ci.org/babel/babylon/jobs/212807733)
* [Node v6 - linux](https://travis-ci.org/babel/babylon/jobs/212807734)
* [Node v7 - linux](https://travis-ci.org/babel/babylon/jobs/212807735)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/212807736)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/212807739)

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213620202)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213620203)

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213765849)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213765850)

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213765788)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/213765789)

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node v4 - linux](https://travis-ci.org/babel/babylon/jobs/214426160)
* [Node v6 - linux](https://travis-ci.org/babel/babylon/jobs/214426163)
* [Node v7 - linux](https://travis-ci.org/babel/babylon/jobs/214426164)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/214426165)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/214426170)

**Comment by hzoo** [comment]
Hey @danez! It looks like one or more of your builds have failed. I've copied the relevant info below to save you some time.

* [Node v4 - linux](https://travis-ci.org/babel/babylon/jobs/214432895)
* [Node v6 - linux](https://travis-ci.org/babel/babylon/jobs/214432896)
* [Node v7 - linux](https://travis-ci.org/babel/babylon/jobs/214432897)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/214432898)
* [Node vlts/* - linux](https://travis-ci.org/babel/babylon/jobs/214432901)

File: src/plugins/flow.js [hunk 1]
```diff
@@ -167,9 +181,94 @@ pp.flowParseDeclareModule = function (node) {
...
+
+  node.kind = kind || "CommonJS";
   return this.finishNode(node, "DeclareModule");
 };
 
+const exportSuggestions = {
+  ["const"]: "declare export var",
+  ["let"]: "declare export var",
+  ["type"]: "export type",
+  ["interface"]: "export interface",
```
Comment (loganfsmyth) [inline src/plugins/flow.js]: Why the computed property?

Comment (danez) [inline src/plugins/flow.js]: Hmm, well...I change it :D

**Pull request merged**